snow/engine/snowman/transitive_test.go

```diff
@@ -93,235 +115,212 @@ func setup(t *testing.T, engCfg Config) (ids.NodeID, validators.Manager, *common
 			return nil, errUnknownBlock
 		}
 	}
 
-	require.NoError(te.Put(context.Background(), vdr, 0, blk.Bytes()))
-
-	vm.ParseBlockF = nil
-
-	require.True(*asked)
-	require.Len(te.blocked, 1)
+	// Attempting to add [child] will cause [parent] to be requested. While the
+	// request for [parent] is outstanding, [child] will be registered into a
+	// job blocked on [parent]'s issuance.
+	require.NoError(engine.Put(context.Background(), peerID, 0, child.Bytes()))
+	require.NotNil(request)
+	require.Len(engine.blocked, 1)
 
 	vm.ParseBlockF = func(context.Context, []byte) (snowman.Block, error) {
 		return nil, errUnknownBytes
 	}
 
-	require.NoError(te.Put(context.Background(), vdr, *reqID, nil))
-
-	vm.ParseBlockF = nil
-
-	require.Empty(te.blocked)
+	// Because this request doesn't provide [parent], the [child] job should be
+	// cancelled.
+	require.NoError(engine.Put(context.Background(), request.NodeID, request.RequestID, nil))
+	require.Empty(engine.blocked)
 }
 
 func TestEngineQuery(t *testing.T) {
 	require := require.New(t)
 
-	vdr, _, sender, vm, te, gBlk := setupDefaultConfig(t)
+	peerID, _, sender, vm, engine := setupDefaultConfig(t)
 
-	blk := &snowman.TestBlock{
-		TestDecidable: choices.TestDecidable{
-			IDV:     ids.GenerateTestID(),
-			StatusV: choices.Processing,
-		},
-		ParentV: gBlk.ID(),
-		HeightV: 1,
-		BytesV:  []byte{1},
-	}
+	blks := BuildChain(Genesis, 2)
+	parent := blks[0]
+	child := blks[1]
 
-	chitted := new(bool)
+	var sendChitsCalled bool
 	sender.SendChitsF = func(_ context.Context, _ ids.NodeID, requestID uint32, preferredID ids.ID, preferredIDByHeight ids.ID, accepted ids.ID) {
-		require.False(*chitted)
-		*chitted = true
+		require.False(sendChitsCalled)
+		sendChitsCalled = true
 		require.Equal(uint32(15), requestID)
-		require.Equal(gBlk.ID(), preferredID)
-		require.Equal(gBlk.ID(), preferredIDByHeight)
-		require.Equal(gBlk.ID(), accepted)
+		require.Equal(GenesisID, preferredID)
+		require.Equal(GenesisID, preferredIDByHeight)
+		require.Equal(GenesisID, accepted)
 	}
 
-	blocked := new(bool)
+	var getBlockCalled bool
 	vm.GetBlockF = func(_ context.Context, blkID ids.ID) (snowman.Block, error) {
-		*blocked = true
+		getBlockCalled = true
+
 		switch blkID {
-		case gBlk.ID():
-			return gBlk, nil
+		case GenesisID:
+			return Genesis, nil
 		default:
 			return nil, errUnknownBlock
 		}
 	}
 
-	asked := new(bool)
-	getRequestID := new(uint32)
-	sender.SendGetF = func(_ context.Context, inVdr ids.NodeID, requestID uint32, blkID ids.ID) {
-		require.False(*asked)
-		*asked = true
-		*getRequestID = requestID
-		require.Equal(vdr, inVdr)
+	var getRequest *common.Request
+	sender.SendGetF = func(_ context.Context, nodeID ids.NodeID, requestID uint32, blkID ids.ID) {
+		require.Nil(getRequest)
+		getRequest = &common.Request{
+			NodeID:    nodeID,
+			RequestID: requestID,
+		}
+		require.Equal(peerID, nodeID)
 		require.Contains([]ids.ID{
-			blk.ID(),
-			gBlk.ID(),
+			parent.ID(),
+			GenesisID,
```

Comment:
aren't parent.ID() and GenesisID equal?

Reply:
`parent.ID()` and `GenesisID` are not equal.